packages/nest/src/generators/utils/normalize-options.ts

```diff
@@ -10,18 +10,19 @@ export async function normalizeOptions(
   tree: Tree,
   options: NestGeneratorOptions
 ): Promise<NormalizedOptions> {
-  const { directory, fileName } =
-    await determineArtifactNameAndDirectoryOptions(tree, {
+  const { artifactName } = await determineArtifactNameAndDirectoryOptions(
+    tree,
+    {
       name: options.name,
       path: options.path,
-    });
+    }
+  );
 
   return {
-    ...options,
     flat: true,
-    name: fileName,
+    name: artifactName,
     skipFormat: options.skipFormat,
-    sourceRoot: directory,
+    sourceRoot: options.path,
```

Comment:
We should get the `directory` from the `determineArtifactNameAndDirectoryOptions` and assign it to the `sourceRoot`. The received `options.path` also contains the filename, so we don't want to provide that as the source root.

I'm doing a similar change in another PR with more general changes, but I'll rebase this one once merged.

```suggestion
    sourceRoot: directory,
```